### PR TITLE
Error handling, logging, and string manipulation.

### PR DESCRIPTION
**Pull request by paulp**

### CODE REVIEW

File: suffuse/opts.go [hunk 1]
```diff
@@ -1,127 +1,107 @@
...
-  fs.BoolVar(&opts.Scratch, "t", false, "create scratch directory as mount point")
-  fs.BoolVar(&opts.Verbose, "v", false, "log at INFO level")
-  fs.BoolVar(&opts.Debug, "d", false, "log at DEBUG level")
-
-  err := fs.Parse(args[1:])
-  if err != nil {
-    // Parse has already printed the problem and usage, there is no way to tell it not to do that
-    return nil, &SfsConfigError{err: err.Error(), usage: nil}
+func optsFromArgs(args []string) *sfsOpts {
+  argMap, err := docopt.Parse(sfsUsage(), args, /*help*/true, /*version*/"HEAD", /*optionsFirst*/false, /*exit*/false)
```
Comment (EECOLOR) [inline suffuse/opts.go:46]: Thank you for putting those comments there

File: suffuse/opts.go [hunk 2]
```diff
@@ -1,127 +1,107 @@
...
+
+Arguments:
+  PATH  existing filesystem
+
+Options:
+  -c PATH     suffuse config file
+  -m PATH     mount point
+  -n VOLNAME  volume name (OS X only)
+  -t          create scratch directory as mount point
+  -v          more verbose logging (also -vv and -vvv)
```
Comment (EECOLOR) [inline suffuse/opts.go:19]: So I guess that library turns `-v` into `1` and `-vv` into `2`?

Comment (paulp) [inline suffuse/opts.go:19]: Right.

File: suffuse/opts.go (hunk 1 above)
Comment (paulp) [inline suffuse/opts.go:46]: Yeah, I have no idea how people come up with such garbage APIs at this stage.

File: suffuse/opts.go (hunk 2 above)
Comment (EECOLOR) [inline suffuse/opts.go:19]: Never mind, figured it out. Btw, that library is pretty amazing

**Comment by EECOLOR** [comment]
Suggested changes in new commit

**Pull request merged**